xdsl/dialects/builtin.py

```diff
@@ -349,7 +349,15 @@ def print_parameter(self, printer: Printer) -> None:
                 raise ValueError(f"Invalid signedness {data}")
 
 
-class FixedBitwidthType(TypeAttribute, ABC):
+class CompileTimeFixedBitwidthType(TypeAttribute, ABC):
+    """
+    A type attribute whose runtime bitwidth is fixed, but may be target-dependent.
+    """
+
+    name = "abstract.compile_time_fixed_bitwidth_type"
+
+
+class FixedBitwidthType(CompileTimeFixedBitwidthType, ABC):
```

Comment:
How do you feel about putting an abstract property here?

Maybe actually not packed-dependent name, like `compile_time_size`?

```suggestion
class CompileTimeFixedBitwidthType(TypeAttribute, ABC):
    """
    A type attribute whose runtime bitwidth is fixed, but may be target-dependent.
    """

    name = "abstract.compile_time_fixed_bitwidth_type"

    @property
    @abstractmethod
    def packed_size(self) -> int:
        """
        Contiguous memory footprint of packed value in bytes.
        """
        raise NotImplementedError()

class FixedBitwidthType(CompileTimeFixedBitwidthType, ABC):
```

Reply:
yep, this makes sense!